python3Packages.{exchange-calendars,pyluach}: init

Tyler-Hardin opened the pull request:
pyluach is a hebrew calendar/date library.

exchange_calendars is a library for exchange holidays/half days/etc. It's a super business day calendar. pyluach is a dependency.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Tyler-Hardin committed:
pyluach: init at 2.2.0

Tyler-Hardin committed:
exchange-calendars: init at 4.9

antonmosich commented on pkgs/development/python-modules/pyluach/default.nix:
```diff
@@ -0,0 +1,35 @@
+  src = fetchFromGitHub {
+    owner = "simlist";
+    repo = "pyluach";
+    rev = "v2.2.0.post1";
```
I think you should use `v2.2.0` here, if you set `version` to `2.2.0`.

antonmosich reviewed (changes requested):
There are a few things you should fix in this:
* Use `dependencies`/`build-system` instead of the older equivalents [1]
* Use `pyproject = true;` instead of `format = "pyproject";` [1]
* If only some checks fail, you should rather just deactivate those instead of setting `doCheck = false;` [2]
* For python packages you can use `pythonImportsCheck` [3]
* You should prefer `tag` over `rev` if possible [4]
* Using `hash` over `sha256` is also more common
* If you use values again (such as `version` in this case) you should use `rec` or the `finalAttrs` pattern if available (which it isn't for `buildPythonPackage` yet afaik)
* `python-packages.nix` should stay sorted as much as possible.

Feel free to ask for clarification should you not understand any of my points.

[1]: https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters
[2]: https://nixos.org/manual/nixpkgs/unstable/#using-pytestcheckhook
[3]: https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck
[4]: https://nixos.org/manual/nixpkgs/unstable/#fetchfromgithub

Tyler-Hardin commented:
@antonmosich thank you for the thorough review! Useful tips. I'm still learning.

I believe I've addressed all your points. Let me know if you need more changes.

antonmosich commented on pkgs/development/python-modules/pyluach/default.nix:
```diff
@@ -0,0 +1,38 @@
+  pythonImportsCheck = [ "pyluach" ];
+
+  meta = with lib; {
+    description = "Library for dealing with Hebrew (Jewish) calendar dates.";
```
`description` [should not end with punctuation](https://nixos.org/manual/nixpkgs/unstable/#var-meta-description)

antonmosich commented on pkgs/top-level/python-packages.nix:
```diff
@@ -11173,6 +11175,8 @@ self: super: with self; {
 
   pkuseg = callPackage ../development/python-modules/pkuseg { };
 
+  pyluach = callPackage ../development/python-modules/pyluach { };
+
   plac = callPackage ../development/python-modules/plac { };
 
```
This still isn't sorted alphabetically, `pyluach` should be further down, shouldn't it?

antonmosich reviewed (commented):
It looks good now, though there are still some minor tweaks you could add:
* remove `with lib;` for e.g. better LSP output (see #208242)
* Add [`meta.changelog`](https://nixos.org/manual/nixpkgs/unstable/#var-meta-changelog) using something like `https://github.com/owner/repo/releases/tag/${src.tag}`
* Change the PR title to better describe what you are doing, "Exchange cal" is quite unclear (see CONTRIBUTING.md)

None of those are particularly important or blocking in my opinion, but are nice to have anyways.

antonmosich commented on pkgs/development/python-modules/pyluach/default.nix:
```diff
@@ -0,0 +1,38 @@
+    owner = "simlist";
+    repo = "pyluach";
+    tag = "v${version}";
+    hash = "sha256-1DmxZsqpRgYaGEiyan6peRwOgQcZ8uuB0nKvRafaCJs=";
```
This hash seems to be wrong, you'll need to update this. This can happen because you have the source with the hash that's currently here in your nix store, and nix doesn't download the source in that case.
